File: binance4j-c2c/src/test/java/com/binance4j/c2c/client/C2CClientTest.java
package com.binance4j.c2c.client;

import org.junit.jupiter.api.Test;

import com.binance4j.c2c.dto.TradeType;
import com.binance4j.c2c.param.TradeHistoryParams;
import com.binance4j.core.exception.ApiException;
import com.binance4j.core.param.Paging;
import com.binance4j.core.test.CustomTest;

public class C2CClientTest extends CustomTest {
	protected Paging paging = new Paging(1, 25);

	protected C2CClient client = new C2CClient(key, secret);

	@Override
	protected C2CClient getClient() {
		return client;
	}

	@Test
	void testGetTrades() throws ApiException {
		testNoNulls(getClient().getTrades(new TradeHistoryParams(TradeType.BUY)));
		testNoNulls(getClient().getTrades(new TradeHistoryParams(TradeType.SELL)));
	}

	@Test
	void testGetTrades2() throws ApiException {
		testNoNulls(getClient().getTrades(new TradeHistoryParams(TradeType.BUY), paging));
		testNoNulls(getClient().getTrades(new TradeHistoryParams(TradeType.SELL), paging));
	}
}

File: src/test/com/binance4j/connectors/c2c/client/C2CClientTest.kt
package com.binance4j.connectors.c2c.client

import com.binance4j.connectors.c2c.dto.TradeType
import com.binance4j.connectors.Connectors
import com.binance4j.connectors.core.test.CustomTest
import org.junit.jupiter.api.Test

internal class C2CClientTest : CustomTest() {
    private var client = Connectors.rest(key, secret).c2c

    @Test
    fun `Get trades by type`() {
        assertDoesNotThrow(client.getTrades(TradeType.BUY))
    }

    @Test
    fun `Get trades by type and page`() {
        assertDoesNotThrow(client.getTrades(TradeType.BUY, 1))
    }

    @Test
    fun `Get trades by type, page and limit`() {
        assertDoesNotThrow(client.getTrades(TradeType.BUY, 1, 25))
    }
}